src/Libraries/Inscriptions/Envelope.ts

```diff
@@ -184,27 +195,31 @@ function getEnvelopeMeta(data: EnvelopeData[]) {
  |--------------------------------------------------------------------------------
  */
 
-function getEnvelopeDataFromTx(tx: RawTransaction): [EnvelopeData[]?, any?] | undefined {
+function getEnvelopesDataFromTx(tx: RawTransaction): [EnvelopeData[]?, any?][] | undefined {
   for (const vin of tx.vin) {
     if (isCoinbase(vin)) {
       continue;
     }
     if (vin.txinwitness) {
-      return getEnvelopeFromTxinWitness(vin.txinwitness);
+      return getEnvelopesFromTxinWitness(vin.txinwitness);
     }
   }
 }
 
-function getEnvelopeFromTxinWitness(txinwitness: string[]): [EnvelopeData[]?, any?] | undefined {
+function getEnvelopesFromTxinWitness(txinwitness: string[]): [EnvelopeData[]?, any?][] | undefined {
   for (const witness of txinwitness) {
     if (witness.includes(PROTOCOL_ID)) {
       const data = script.decompile(Buffer.from(witness, "hex"));
       if (data) {
         const oip = getMetaFromWitness(txinwitness);
         if (oip) {
-          return [getEnvelope(data), oip];
+          return getEnvelopes(data).map((envelope) => {
+            return [envelope, oip];
+          });
         }
-        return [getEnvelope(data)];
+        return getEnvelopes(data).map((envelope) => {
+          return [envelope];
+        });
```

Comment:
hmm an ordzaar inscription will now looks like `[[[inscriptioni0, oip]], [inscriptioni1, oip]]]` right?
because now the oip inscription will be counted and second inscription 🤔

Reply:
ohh you're right, i'll check the oip data position first

Reply:
just checked, the inscriptions will be considered as oip if one of the inscription has type json and the oip field, example: 
```
<inscription_1_image>,
<inscription_2_image>,
<inscription_3_valid_oip_json>
```
in that example all of the inscriptions will be injected the oip metadata also

Reply:
ya oip (second inscription) will be indexed bcs it's a valid inscription, it also will appear on the profile @thedoublejay

Reply:
Yep I don't mind showing this. Because it's the behavior anyway on other wallets